src/pcre2_compile.c

```diff
@@ -4287,11 +4279,15 @@ while (ptr < ptrend)
             }
 
           ptr++;
+          *parsed_pattern++ = META_SCS;
           /* Temporary variable, zero in the first iteration. */
           meta = 0;
```

Comment:
You've copy-pasted the code for writing out the META_OFFSET twice, once in each branch (number/name). So maybe do it a third time by getting rid of the `meta` variable, and hoist out the initial write of the META_OFFSET to the top?

Reply:
I have reworked the code.

Reply:
Even better now, thank you!